### PR TITLE
Flask API: Improve installation

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR closes https://github.com/wazuh/wazuh/issues/2568.

Now the `api` python package is only installed under `/var/ossec/framework/python/lib/python3.7/site-packages/` and not under `/var/ossec/api`. Unit tests aren't installed either.

Best regards,
Marta

### CODE REVIEW

File: api/Makefile [hunk 1]
```diff
@@ -11,24 +11,16 @@ RM_FILE      = rm -f
 INSTALL_DIR  = install -o root -g ${OSSEC_GROUP} -m 0750  -d
```
Comment (crd1985) [inline api/Makefile]:
```suggestion
INSTALL_DIR  = install -o root -g ${OSSEC_GROUP} -m 0750  -d
INSTALL_RW_DIR  = install -o root -g ${OSSEC_GROUP} -m 0770  -d
```

File: api/Makefile [hunk 2]
```diff
@@ -11,24 +11,16 @@ RM_FILE      = rm -f
...
 .PHONY: all install
 
 all: install
 
 install:
-	$(INSTALL_DIR) $(PREFIX)/api/api
-	$(INSTALL_DIR) $(PREFIX)/api/api/controllers
-	$(INSTALL_DIR) $(PREFIX)/api/api/models
-	$(INSTALL_DIR) $(PREFIX)/api/api/spec
 	$(INSTALL_DIR) $(PREFIX)/api/configuration
```
Comment (crd1985) [inline api/Makefile]:
```suggestion
	$(INSTALL_RW_DIR) $(PREFIX)/api/security
	$(INSTALL_DIR) $(PREFIX)/api/configuration
```

File: api/scripts/wazuh-apid.py [hunk 3]
```diff
@@ -6,13 +6,12 @@
...
 import os
 import sys
-import yaml
 import ssl
 
 import connexion
 from flask_cors import CORS
 
-from api import encoder, configuration
+from api import encoder, configuration, __path__ as api_path
```
Comment (crd1985) [inline api/scripts/wazuh-apid.py]: Just a suggestion to check if the following links fit better what we want achieve here:
* https://stackoverflow.com/questions/247770/how-to-retrieve-a-modules-path
* https://stackoverflow.com/questions/5183601/how-to-get-the-current-running-module-path-name/8789689

Comment (mgmacias95) [reply api/scripts/wazuh-apid.py]: I think the `__path__` variable fits our needs. From the [docs](https://docs.python.org/3.7/tutorial/modules.html#packages-in-multiple-directories):
> Packages support one more special attribute, `__path__`. This is initialized to be a list containing the name of the directory holding the package’s` __init__.py` before the code in that file is executed. This variable can be modified; doing so affects future searches for modules and subpackages contained in the package.

File: api/Makefile (hunk 2 above)
Comment (mgmacias95) [reply api/Makefile]: Added in 2fc48a8612adcf3319996e315cd6fa801500548a

File: api/Makefile (hunk 1 above)
Comment (mgmacias95) [reply api/Makefile]: Added in 2fc48a8612adcf3319996e315cd6fa801500548a.

File: api/scripts/wazuh-apid.py [hunk 4]
```diff
@@ -31,7 +30,7 @@ def print_version():
 
 
 def main(cors, port, host, ssl_context, main_logger):
-    app = connexion.App(__name__, specification_dir=f'{common.ossec_path}/api/api/spec/')
+    app = connexion.App(__name__, specification_dir=os.path.join(api_path[0], 'spec/'))
```
Comment (crd1985) [inline api/scripts/wazuh-apid.py]:
```suggestion
    app = connexion.App(__name__, specification_dir=os.path.join(api_path[0], 'spec'))
```

**Pull request merged**